components/modal/modal.html

```diff
@@ -0,0 +1,142 @@
+                                ad non.</p>
+                        </div>
+                        <div class="modal_footer flex justify_end">
+                            <button class="btn btn_outlined" id="modal_close_btn">Cancel</button>
+                            <button class="btn btn_primary">Ok</button>
+                        </div>
+                    </div>
+                </div>
+                <br />
+                <div class="code_snippet relative_pos">
+                    <iframe
+                        src="https://carbon.now.sh/embed?bg=rgba%28187%2C187%2C187%2C1%29&t=one-light&wt=sharp&l=htmlmixed&width=680&ds=true&dsyoff=3px&dsblur=13px&wc=true&wa=true&pv=26px&ph=16px&ln=false&fl=1&fm=Hack&fs=14px&lh=133%25&si=false&es=2x&wm=false&code=%253Cdiv%2520id%253D%2522modal_container%2522%2520class%253D%2522modal%2522%253E%250A%2509%253Cdiv%2520class%253D%2522modal-content%2522%253E%250A%2509%2509%253Cdiv%2520class%253D%2522modal_title%2522%253E%250A%2509%2509%2509%253Ch3%253EModal%2520Title%253C%252Fh3%253E%250A%2509%2509%253C%252Fdiv%253E%250A%2509%2509%253Cdiv%2520class%253D%2522modal_body%2520py-2%2522%253E%250A%2509%2509%2509%253Cp%253E%250A%2520%2520%2520%2520%2520%2520%2509%2520%2520%2520%2520%2509Some%2520text%2520in%2520the%2520Modal..%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2509Eiusmod%2520fugiat%2520sit%2520eu%2520elit%2520officia%2520eu%2520veniam%2520exercitation%2520veniamad%2520non.%2509%2520%2520%2520%250A%2520%2520%2520%2520%2520%2520%2520%2509%2520%2520%2520%2520%253C%252Fp%253E%250A%2509%2509%253C%252Fdiv%253E%250A%2509%2509%253Cdiv%2520class%253D%2522modal_footer%2520flex%2520justify_end%2522%253E%250A%2509%2509%2509%253Cbutton%2520class%253D%2522btn%2520btn_outlined%2522%2520id%253D%2522modal_close_btn%2522%253ECancel%253C%252Fbutton%253E%250A%2509%2509%2509%253Cbutton%2520class%253D%2522btn%2520btn_primary%2522%253EOk%253C%252Fbutton%253E%250A%2509%2509%253C%252Fdiv%253E%250A%2509%253C%252Fdiv%253E%250A%253C%252Fdiv%253E"
+                        style="width: 837px; height: 443px; border:0; transform: scale(1); overflow:hidden;"
+                        sandbox="allow-scripts allow-same-origin" id="iframe">
+                    </iframe>
+                </div>
+            </section>
+        </main>
+    </div>
+</body>
+<script>
+    const modal = document.getElementById("modal_container");
+    const openBtn = document.getElementById("modal_open_btn");
+    const closeBtn = document.getElementById("modal_close_btn");
+    openBtn.onclick = function () {
+        modal.style.display = "block";
+    }
+    closeBtn.onclick = function () {
+        modal.style.display = "none";
+    }
+
+    // When the user clicks anywhere outside of the modal, close it
+    window.onclick = function (event) {
+        if (event.target == modal) {
+            modal.style.display = "none";
+        }
+    }
+</script>
+<script src="../../javascript/toggle.js"></script>
```

Comment:
you can also use "defer" attribute with script tags and put script on top so file looks clean